src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuizExerciseResource.java

```diff
@@ -309,6 +330,249 @@ public QuizExerciseResource(QuizExerciseRepository quizExerciseRepository, Parti
+                if (submittedAnswer instanceof MultipleChoiceSubmittedAnswer) {
+                    // Delete all references to question an answers if the question was deleted
+                    if (!quizExercise.getQuestions().contains(submittedAnswer.getQuestion())) {
+                        submittedAnswer.setQuestion(null);
+                        ((MultipleChoiceSubmittedAnswer) submittedAnswer).setSelectedOptions(null);
+                        submittedAnswersToDelete.add(submittedAnswer);
+                    } else {
+                        // find same question in quizExercise
+                        for (Question question : quizExercise.getQuestions()) {
+                            if (question.getId().equals(submittedAnswer.getQuestion().getId())) {
+                                // Check if an answerOption was deleted and delete reference to in selectedOptions
+                                Set<AnswerOption> selectedOptionsToDelete = new HashSet<>();
+                                for (AnswerOption answerOption : ((MultipleChoiceSubmittedAnswer) submittedAnswer).getSelectedOptions()) {
+                                    if (!((MultipleChoiceQuestion) question).getAnswerOptions().contains(answerOption)) {
+                                        selectedOptionsToDelete.add(answerOption);
+                                    }
+                                }
+                                ((MultipleChoiceSubmittedAnswer) submittedAnswer).getSelectedOptions().removeAll(selectedOptionsToDelete);
+
+                            }
+                        }
+                    }
+                }
+                // TO-DO DragAndDrop Question
+            }
+            ((QuizSubmission) result.getSubmission()).getSubmittedAnswers().removeAll(submittedAnswersToDelete);
+            quizSubmissionRepository.save((QuizSubmission) result.getSubmission());
+        }
+
+        //update QuizExercise
+        ResponseEntity<QuizExercise> methodResult = updateQuizExercise(quizExercise);
+
+        // update Statistics and Results
+        if (updateOfResultsAndStatisticsNecessary) {
+
+            updateStatisticsAndResults(quizExercise);
+        }
+
+        return methodResult;
+    }
+
+    /**
+     * 1. undo all changes which are not allowed ( dueDate, releaseDate, question.points, adding Questions and Answers)
+     * 2. check if an update of the Results and Statistics is necessary
+     *
+     * @param quizExercise the changed QuizExercise object
+     * @param originalQuizExercise the original QuizExercise object which will be compared with the new changed quizExercise
+     *
+     * @return a boolean which is true if an update is necessary and false if not
+     */
+    private boolean undoUnallowedChangesAndCheckIfRecalculationIsNecessary (QuizExercise quizExercise, QuizExercise originalQuizExercise){
+
```

Comment:
This method is till too complex in my opinion. If you write good and clean code, a method should only have one very specific purpose and should NOT include more than 15-20 lines of code (see https://dzone.com/articles/rule-30-%E2%80%93-when-method-class-or)

In addition, you have nested "for loops" with 4 levels here, which is unnecessary complex.
Please avoid nested loops with more than 2 levels in one method. You can e.g. write a helper method that finds the corresponding question in the original quiz exercise and another one that finds the corresponding answer options in the original question. Then you would reduce the complexity. I would suggest that you create a method findQuestion(Long questionId) in QuizExercise and that you write another method findAnswerOption(Long answerOptionId) in QuizQuestion. 

Then please review your whole code and whenever you need to find a question or an answer option, please use these methods!